internal/backup.go

```diff
@@ -400,15 +433,25 @@ func GetBackupsAndGarbageWithTarget(folder storage.Folder, targetPath string) (b
 		return nil, nil, err
 	}
 
-	sortTimes := GetBackupTimeSlices(backupObjects)
+	sortTimes, err := GetBackupTimeSlices(backupObjects, folder, CreationTime)
+	switch err.(type) {
+	case FailedMetadataError:
+		sortTimes, err = GetBackupTimeSlices(backupObjects, folder, ModifyTime)
+	}
+	if err != nil {
+		return nil, nil, err
+	}
 	garbage = getGarbageFromPrefix(subFolders, sortTimes)
 
 	return sortTimes, garbage, nil
 }
 
 // TODO : unit tests
-func GetBackupTimeSlices(backups []storage.Object) []BackupTime {
+func GetBackupTimeSlices(backups []storage.Object, folder storage.Folder, order BackupOrder) ([]BackupTime, error) {
 	sortTimes := make([]BackupTime, len(backups))
+	if order != CreationTime && order != ModifyTime {
```

Comment:
We may need the backup sorting logic in the other places later, maybe introduce an interface like `BackupSorter` or `BackupComparator` and pass it to `GetBackupTimeSlices`?